src/transformState.js

```diff
@@ -2,10 +2,31 @@
 
 /**
  * @param {Object} state
+ * @typeof {Object} Action
+ * @property {string} type
+ * @property {Object} extraData
+ * @property {string[]} keysToRemove
  * @param {Object[]} actions
  */
+
+const ADD_PROPERTIES = 'addProperties';
+const REMOVE_PROPERTIES = 'removeProperties';
+const CLEAR = 'clear';
+
 function transformState(state, actions) {
-  // write code here
+  actions.forEach((action) => {
+    if (action.type === ADD_PROPERTIES) {
+      Object.assign(state, action.extraData);
+    } else if (action.type === REMOVE_PROPERTIES) {
+      action.keysToRemove.forEach((key) => {
+        delete state[key];
+      });
+    } else if (action.type === CLEAR) {
+      Object.keys(state).forEach((key) => {
+        delete state[key];
+      });
+    }
```

Comment:
according to [checklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) you should use switch statement